src/richText.js

```diff
@@ -277,11 +277,40 @@ define([
             }
         });
 
-        // Adapted from http://www.keyvan.net/2012/11/clean-up-html-on-paste-in-ckeditor/
-        var STYLE = /<style type="text\/css">.*?<\/style>/g;
+        editor._vellum_fromRichText = function (html) {
+            return fromRichText(html, form, options.isExpression);
+        };
+
         editor.on('paste', function(event) {
-            event.data.dataValue = event.data.dataValue.replace(STYLE, "");
-        }, null, null, 2 );
+            var data = event.data;
+            if (data.dataTransfer && data.dataTransfer.getData("Text")) {
+                // Get plain text instead of HTML because HTML encoded
+                // content from applications like Word or your text
+                // editor often contains unwanted styling information.
+                //
+                // Insert HTML-ified plain text rather than HTML with
+                // bubbles because we cannot reliably find hashtags in
+                // text fragments that are not valid XPath expressions.
```

Comment:
It's been a while since I've looked at this code, but doesn't `toRichText` attempt to do something like this smartly? I think the right workflow might be something like

- attempt to process string as an xpath with hashtags (with or without the escaping `)
- if that fails, look for escaped hashtags (ones with `)
- if that fails assume it's a string and shouldn't be parsed